pkg/auth/user/repository/helper/UserRepositoryQueryBuilder.go

```diff
@@ -0,0 +1,85 @@
+package helper
+
+import (
+	"fmt"
+	"github.com/devtron-labs/devtron/api/bean"
+	bean2 "github.com/devtron-labs/devtron/pkg/auth/user/bean"
+	"go.uber.org/zap"
+	"strconv"
+)
+
+type UserRepositoryQueryBuilder struct {
+	logger *zap.SugaredLogger
+}
+
+func NewUserRepositoryQueryBuilder(logger *zap.SugaredLogger) UserRepositoryQueryBuilder {
+	userListingRepositoryQueryBuilder := UserRepositoryQueryBuilder{
+		logger: logger,
+	}
+	return userListingRepositoryQueryBuilder
+}
+
+func (impl UserRepositoryQueryBuilder) GetQueryForUserListingWithFilters(req *bean.FetchListingRequest) string {
+	whereCondition := fmt.Sprintf("where active = %t AND (user_type is NULL or user_type != '%s') ", true, bean.USER_TYPE_API_TOKEN)
+	orderCondition := ""
+
+	if len(req.SearchKey) > 0 {
+		emailIdLike := "%" + req.SearchKey + "%"
+		whereCondition += fmt.Sprintf("AND email_id ilike '%s' ", emailIdLike)
+	}
+
+	if len(req.SortBy) > 0 && req.Size > 0 {
+		orderCondition += fmt.Sprintf("order by %s ", req.SortBy)
+		if req.SortOrder == bean2.Desc {
+			orderCondition += string(req.SortOrder)
+		}
+	}
+
+	if req.Size > 0 {
+		orderCondition += " limit " + strconv.Itoa(req.Size) + " offset " + strconv.Itoa(req.Offset) + ""
+	}
+	var query string
+	if req.Size == 0 {
+		query = fmt.Sprintf("select count(*) from users AS user_model left join user_audit AS au on au.user_id=user_model.id %s %s;", whereCondition, orderCondition)
+	} else {
+		// have not collected client ip here. always will be empty
+		query = fmt.Sprintf("SELECT \"user_model\".*, \"user_audit\".\"id\" AS \"user_audit__id\", \"user_audit\".\"updated_on\" AS \"user_audit__updated_on\",\"user_audit\".\"user_id\" AS \"user_audit__user_id\" ,\"user_audit\".\"created_on\" AS \"user_audit__created_on\" ,\"user_audit\".\"updated_on\" AS \"last_login\" from users As \"user_model\" LEFT JOIN user_audit As \"user_audit\" on \"user_audit\".\"user_id\" = \"user_model\".\"id\" %s %s;", whereCondition, orderCondition)
```

Comment:
use single quote for string declaration and remove escaping slashes

Reply:
DONE